### ISSUE TITLE
Isort I001 spuriously treats package sometimes as first-party sometimes as third party if submodule has the same name as the package.

### ISSUE THREAD

**Issue by Daraan** [astral-sh/ruff#21202]
### Summary

I have a subpackage with file called `module.py` in this file I import the third-party package `module` which has the same name. The `I`sort rules move this package into the first party block.

```python
# my_package/subpackage/module.py

#  ---- Third party imports ----

import numpy as np
# from module import something # <-- should be here, but gets moved into first party block.

# First party block
from my_package import foo 
from module import something # <--- isort ends up here. Should be "module" package not this file
````

### Version

0.14.3

**Comment by dylwil3** [comment]
Thanks for the report! Can you help me reproduce this? Here was my failed attempt:

```bash
❯ uv init mypackage --lib \
  && cd mypackage \
  && mkdir src/mypackage/subpackage \
  && touch src/mypackage/subpackage/__init__.py \
  && echo 'from module import something' > src/mypackage/subpackage/module.py

Initialized project `mypackage` at `/Users/dmbp/Documents/dev/repros/mypackage`

❯ ruff check --no-cache --isolated --select I001 -v
[... omitted various logs...]
[2025-11-02][07:00:58][ruff_linter::rules::isort::categorize][DEBUG] Categorized 'module' as Known(ThirdParty) (NoMatch)
All checks passed!
```

**Comment by Daraan** [comment]
Best I share the project an file: https://github.com/Daraan/ray_utilities/blame/dev/WIP/ray_utilities/callbacks/wandb.py  (not main branch).

However, this is really strange when I wrote the PR running `ruff check ray_utilities/callbacks/wandb.py --select I001  --line-length 120` moved the imports down. Now, same file same command, same ruff version it moves the imports that ruff moved down itself up again. Normally I use the ruff VSCode ruff extension to sort the imports. But for this PR I updated ruff and ran it with the cli to make sure its not the extension.

I already noticed this strange pattern in my commits as well:

If I would commit now:
(move up)

https://github.com/Daraan/ray_utilities/blame/99ebd9234074529eea05454aa1cdf7eccc40fc50/ray_utilities/callbacks/wandb.py  (moved down)

prior blame:
https://github.com/Daraan/ray_utilities/blame/4127cf2904c90dd12c8fabe20e31d3fc97c994da/ray_utilities/callbacks/wandb.py (moved up)

blame before
https://github.com/Daraan/ray_utilities/blame/57e7c372e97c1302297bda3c19b6c5873fa218e5/ray_utilities/callbacks/wandb.py (moved down)

original implementation
https://github.com/Daraan/ray_utilities/blame/6d34712a879e1205a4b04bf1001724b824ced70e/ray_utilities/callbacks/wandb.py
(up)

---

I guess something spurious is happening that always dislikes the current state

**Comment by dylwil3** [comment]
I'm away from my computer at the moment but I suspect this has to do with the non-python folder that `wandb` sometimes creates inside of the package that's called `wandb` (and is read as a namespace package). This has come up before and it is on my to-do list to add `wandb` as a default `lint.isort.known-third-party` but I haven't gotten a chance! 

Can you see if adding that to your Ruff config solves the issue in the meantime?

**Comment by Daraan** [comment]
Normally I log wandb elsewhere, but in fact I do have a `wandb` folder around. I added it to known-third-party.

Currently `--isolated` then wants to move it up, without ruff is (currently) happy when it is down at first-party. I now expected that `--isolated` moves it down as it sees a `wandb` folder and respecting `pyproject.toml` would then move it up. Did I twist my thinking?
